### DISCUSSION TITLE
Custom field drafts

### DISCUSSION THREAD

**Discussion by MrMalkio** [joomla/joomla-cms#36045] [2. Cold Features]
Being as though fields are locked with the original field type, it would be beneficial to have a setting that makes a field a draft before solidifying it as a field with that spefic name and type. 

This is useful in times when the content structure is still being crafted, accounts for mistakes of field type etc. 

User would be able to save a custom field as a draft or save as official field

**Comment by brianteeman** [comment]
I understand what you mean but how would you address the fact that the options for the field will change depending on the field type selected.

**Reply by MrMalkio** [reply]
@brianteeman I think a simple modal would suffice in the first version of this, that warns users of how the change would be affected, and in some cases, it could even do a "best-effort" change when switching to the new field type by taking the already populated options to the next type.

For example Checkbox, radio, list, & imagelist should be easily interchangeable as they are basically the same thing with a 1 to 2 option difference.  And the same goes for Text, text, editor area, and URL they are just more or less advanced versions of each other. 

Furthermore, the feature could have tags attached to the new field options that only show after a draft is saved or the options to it have been filled in, that lets you know what, I think the issue with options changing is one of the reasons that drafting would be beneficial.

If we had the ability to draft, then we could also comfortably use the new j4 API to make field updates en masse/bulk. For example, right now I have to add over 200 custom article fields and just under 200 custom user fields. I would use the API to do so if I was confident of each type already but I'm not and I must go 1 by 1.  with this same thought, I think drafts would also further empower the new workflow feature.

And lastly, if drafts were available then we could also use the "Save as draft copy" to make fields faster, that use the same status, field group, categories, Access, and note field content but not the same field type.

**Reply by brianteeman** [reply]
that's a very specific use case.
